src/rounded_rect_radii.rs

```diff
@@ -0,0 +1,104 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A description of the radii for each corner of a rounded rectangle.
+
+use std::convert::From;
+
+/// Radii for each corner of a rounded rectangle.
+#[derive(Clone, Copy, Default, Debug, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct RoundedRectRadii {
+    /// The radius of the top-left corner.
+    pub top_left: f64,
+    /// The radius of the top-right corner.
+    pub top_right: f64,
+    /// The radius of the bottom-right corner.
+    pub bottom_right: f64,
+    /// The radius of the bottom-left corner.
+    pub bottom_left: f64,
+}
+
+impl RoundedRectRadii {
+    /// Create a new RoundedRectRadii. This function takes radius values for
+    /// the four corners. The argument order is "top_left, top_right,
+    /// bottom_right, bottom_left", or clockwise starting from top_left.
+    pub const fn new(top_left: f64, top_right: f64, bottom_right: f64, bottom_left: f64) -> Self {
+        RoundedRectRadii {
+            top_left,
+            top_right,
+            bottom_right,
+            bottom_left,
+        }
+    }
+
+    /// Create a new RoundedRectRadii from a single radius. The `radius`
+    /// argument will be set as the radius for all four corners.
+    pub const fn from_single_radius(radius: f64) -> Self {
+        RoundedRectRadii {
+            top_left: radius,
+            top_right: radius,
+            bottom_right: radius,
+            bottom_left: radius,
+        }
+    }
+
+    /// Takes the absolute value of all corner radii.
+    pub fn abs(&self) -> Self {
+        RoundedRectRadii::new(
+            self.top_left.abs(),
+            self.top_right.abs(),
+            self.bottom_right.abs(),
+            self.bottom_left.abs(),
+        )
+    }
+
+    /// For each corner, takes the min of that corner's radius and `max`.
+    pub fn clamp(&self, max: f64) -> Self {
+        RoundedRectRadii::new(
+            self.top_left.min(max),
+            self.top_right.min(max),
+            self.bottom_right.min(max),
+            self.bottom_left.min(max),
+        )
+    }
+
+    /// Returns true if all radius values are finite.
+    pub fn is_finite(&self) -> bool {
+        self.top_left.is_finite()
+            && self.top_right.is_finite()
+            && self.bottom_right.is_finite()
+            && self.bottom_left.is_finite()
+    }
+
+    /// Returns true if any corner radius value is NaN.
+    pub fn is_nan(&self) -> bool {
+        self.top_left.is_nan()
+            || self.top_right.is_nan()
+            || self.bottom_right.is_nan()
+            || self.bottom_left.is_nan()
+    }
+}
```

Comment:
I think it would also be useful to have `as_single_radius(&self) -> Option<f64>` that succeeds when the radii are all equal. There may well be more efficient rendering paths when that is the case.